hnix-store-remote/src/System/Nix/Store/Remote.hs

```diff
@@ -156,6 +158,23 @@ buildPaths ps bm = do
     putPaths ps
     putInt $ fromEnum bm
 
+buildDerivation :: StorePath
+                -> Derivation StorePath Text
+                -> BuildMode
+                -> MonadStore BuildResult
+buildDerivation p drv buildMode = do
+  runOpArgs BuildDerivation $ do
+    putPath p
+    putDerivation drv
```

Comment:
This design decision bugs me in the nix protocol. At that point, the .drv is already present in the store. Why not provide a `buildDerivation(/nix/store/xxx.drv)` ? I guess the reason is about garbage collection and such, but still. What is the point of writing .drv's to the store then ?

Reply:
Or rather, why do we have to upload both the path and the drv itself ?

Reply:
Never mind, I found the wopBuildPaths that does exactly that. wopBuildDerivation is pretty specific for "anonymous" builds and was introduced for hydra in https://github.com/NixOS/nix/commit/1511aa9f488ba0762c2da0bf8ab61b5fde47305d

Reply:
Thanks for the commit link. I was a bit confused about that as well.